blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -5,6 +5,7 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.net.Socket;
+import java.nio.ByteBuffer;
```

Comment:
I think I would take a look in the debugger at the content of the byte array you are sending before and after patch to see if you can work out why the byte[] is different.

If making the change just that syncing the outputStream would work I would consider that too initially. We haven't looked into.

Finally. does _not_ allocating a StringBuffer and just allocating a bytebuffer (which you need to do anyway) and copying in to that work?

Last observation is that my inspection of the patch also concurs it looks functionally equivalent. I wonder if there is a bug in the test that is exposed by closing this threading issue?